packages/zero-runtime/src/processors/useThemeProps.ts

```diff
@@ -0,0 +1,73 @@
+import { validateParams, IOptions as IBaseOptions } from '@linaria/tags';
+import type { Expression, Params, TailProcessorParams } from '@linaria/tags';
+import BaseProcessor from './base-processor';
+import { valueToLiteral } from '../utils/valueToLiteral';
+
+type IOptions = IBaseOptions & {
+  themeArgs: {
+    theme: { components?: Record<string, { defaultProps?: Record<string, unknown> }> };
+  };
+};
+
+export class UseThemePropsProcessor extends BaseProcessor {
+  componentName: string;
+
+  constructor(params: Params, ...args: TailProcessorParams) {
+    super(params, ...args);
+    if (params.length > 2) {
+      // no need to do any processing if it is an already transformed call or just a reference.
+      throw BaseProcessor.SKIP;
+    }
+    validateParams(params, ['callee', 'call'], `Invalid use of ${this.tagSource.imported} tag.`);
+    const [, callParam] = params;
+    const [, callArg] = callParam;
+    if (!callArg || callArg.ex.type !== 'StringLiteral') {
+      throw new Error(
+        `Invalid usage of \`createUseThemeProps\` tag, expected one string literal argument but got ${callArg?.ex.type}.`,
+      );
+    }
+    this.componentName = callArg.ex.value;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  build(): void {}
```

Comment:
You might have to add a css comment. Can you verify the build in the demo vite and next app?

Reply:
The build looks fine to me. I got no error.